### PR TITLE
NBSNEBIUS-65: add rdma documentation

### PR DESCRIPTION
**Pull request by budevg**

### CODE REVIEW

File: doc/rdma/architecture.md [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+`Read/WriteDeviceBlocksRequest`.
+
+For `WriteDeviceBlocksRequest`, it constructs `WriteBlocksRequest` using the
+same data buffers and writes data using `StorageAdaptor`.
+
+For `ReadDeviceBlocksRequest`, it uses `StorageAdaptor` to read data into
+`ReadBlocksRequest` and then converts response to `ReadDeviceBlocksRequest`.
+
+There is extra buffer copy inside the `StorageAdaptor`. We use aio for
+read/write and it require page_size aligned buffers. When we call
```
Comment (0x2493) [inline doc/rdma/architecture.md:84]: requires

File: doc/rdma/architecture.md [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+RDMA library provides interface to communicate
+through the RDMA transport. The interface is composed of two entities: `Client`
+and `Server`. It allows multiple Clients to connect to a single Server.
+
+### [Client](../../cloud/blockstore/libs/rdma/iface/client.h)
+
+Single client object (created using `NRdma::CreateClient`) can connect to multiple
+servers using `StartEndpoint`. After successful connection `Endpoint` can be
+used to allocate request (`AllocateRequest`) and sending it (`SendRequest`).
```
Comment (0x2493) [inline doc/rdma/architecture.md:12]: sending -> send

File: doc/rdma/architecture.md [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+
+Each endpoint pre-allocates `QueueDepth` buffers for request/response
+metadata. Additional buffers for request/response data is allocated on demand
```
Comment (0x2493) [inline doc/rdma/architecture.md:15]: is -> are

**Review by 0x2493 — APPROVED** [review]

**Comment by 0x2493** [comment]
Looks good to me except for some typos. I'll go over the doc later and fix them

**Review by qkrorlqr — APPROVED** [review]

**Pull request merged**